File: go/book.go
package main

// Book 本の構造体
type Book struct {
	id     int    `csv:"id"`
	name   string `csv:"name"`
	status int    `csv:"status"`
}

func newBook(id int, name string, status int) *Book {
	b := new(Book)
	b.id = id
	b.name = name
	b.status = status
	return b
}

File: src/book.rb
# coding: UTF-8
class Book

  STATUS_RESERVED = 1
  STATUS_MULTIPLE = 2
  STATUS_ERROR    = 3
  STATUS_NO_BOOK  = 4

  def initialize(status, name)
    @status = status
    @name = name
  end

  def getName
    return @name
  end

  def getStatus
    return @status
  end
end
